pkg/registry/core/service/storage/rest.go

```diff
@@ -115,7 +126,10 @@ func NewREST(
 		proxyTransport:      proxyTransport,
 		pods:                pods,
 	}
-
+	// Configure the after delete func for both service and service status storage to ensure
```

Comment:
This feels like a strange approach which isn't clearly documented.

You are defining a new type, then calling one method on it, passing the same argument to both types.  What purpose does the ServiceStatusStorage type serve?  Why not just set both AfterDelete functions in one method on one type?

I feel like there must be a reason, and I just don't see it.

Reply:
Agreed that this is strange. I have removed the unnecessary types and passing in an explicit `setAfterDelete` function so that it can set for both storages. PTAL if this is cleaner, thanks!